dhcpv4/ztpv4/ztp.go

```diff
@@ -15,14 +17,36 @@ type VendorData struct {
 
 var errVendorOptionMalformed = errors.New("malformed vendor option")
 
-// ParseVendorData will try to parse dhcp4 options looking for more
-// specific vendor data (like model, serial number, etc).
-func ParseVendorData(packet *dhcpv4.DHCPv4) (*VendorData, error) {
+func parseVIVC(packet *dhcpv4.DHCPv4) (*VendorData, error) {
+	vd := &VendorData{}
+	for _, id := range packet.VIVC() {
+		if id.EntID == uint32(iana.EntIDCiscoSystems) {
+			vd.VendorName = iana.EntIDCiscoSystems.String()
+			//SN:0;PID:R-IOSXRV9000-CC
+			for _, f := range bytes.Split(id.Data, []byte(";")) {
+				p := bytes.Split(f, []byte(":"))
+				if len(p) != 2 {
+					return nil, errVendorOptionMalformed
+				}
+
+				switch string(p[0]) {
+				case "SN":
+					vd.Serial = string(p[1])
+				case "PID":
+					vd.Model = string(p[1])
+				}
+			}
+		}
+	}
+	return vd, nil
+}
+
+func parseVendorClass(packet *dhcpv4.DHCPv4) (*VendorData, error) {
```

Comment:
maybe call this `parseClassIdentifier` to match the packet method and to be consistent with the `parseVIVC` function.